src/IO/S3/copyS3File.cpp

```diff
@@ -435,22 +440,21 @@ namespace
             /// Suppress warnings because bg_tasks_mutex is actually hold, but tsa annotations do not understand std::unique_lock
             bg_tasks_condvar.wait(lock, [this]() {return TSA_SUPPRESS_WARNING_FOR_READ(num_added_bg_tasks) == TSA_SUPPRESS_WARNING_FOR_READ(num_finished_bg_tasks); });
 
-            auto & tasks = TSA_SUPPRESS_WARNING_FOR_WRITE(bg_tasks);
-            for (auto & task : tasks)
```

Comment:
The problem was here. A multipart upload means that the whole file is splitted into parts, and for each part there is task which uploads it. Tasks are executed in parallel. If a tasks fails, the whole uploading is aborted. That means other tasks can produce that exception
```
The specified upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed
```

This loop here iterated through all teh tasks and rethrew the first found exception. Which could be either the actual error or this meaningless message `The specified upload does not exist`.

In order to fix that it's enough to keep and then rethrow only the first thrown exception - while uploading one file (any following exceptions don't make much sense).